packages/faceted-search/src/components/Badges/BadgeTags/BadgeTagsForm.component.js

```diff
@@ -0,0 +1,184 @@
+import React, { useState, useCallback, useMemo } from 'react';
+import PropTypes from 'prop-types';
+import get from 'lodash/get';
+import Action from '@talend/react-components/lib/Actions/Action';
+import FilterBar from '@talend/react-components/lib/FilterBar';
+import RichLayout from '@talend/react-components/lib/RichTooltip/RichLayout';
+import Loader from '@talend/react-components/lib/Loader';
+import { CIRCULAR_PROGRESS_SIZE } from '@talend/react-components/lib/constants';
+import { Checkbox } from '@talend/react-components/lib/Toggle';
+import { getTheme } from '@talend/react-components/lib/theme';
+import cssModule from './BadgeTags.scss';
+import { getApplyDataFeature } from '../../../helpers/usage.helpers';
+
+const theme = getTheme(cssModule);
+
+const BadgeTag = ({ checked, id, label, onChange }) => {
+	const describedby = `${id}-${label}`;
+	const onChangeTag = event => {
+		onChange(event, id);
+	};
+	return (
+		<React.Fragment>
+			<Checkbox
+				onChange={onChangeTag}
+				aria-describedby={describedby}
+				id={`${id}-checkbox`}
+				label={label}
+				checked={checked}
+			/>
+			<div id={describedby} className="sr-only">
+				{label}
+			</div>
+		</React.Fragment>
+	);
+};
+
+BadgeTag.propTypes = {
+	id: PropTypes.string.isRequired,
+	label: PropTypes.string.isRequired,
+	onChange: PropTypes.func.isRequired,
+	checked: PropTypes.bool,
+};
+
+const createTagEntity = value => checkbox => {
+	const entity = value.find(v => v.id === checkbox.id);
+	return {
+		id: checkbox.id,
+		label: checkbox.label,
+		checked: entity ? entity.checked : checkbox.checked || false,
+	};
+};
+
+const getTags = (tagsValues, value) => {
+	return tagsValues.map(createTagEntity(value));
+};
+
+const getVisibleTags = (tags, filterValue, showAll) => {
+	const formatFilterValue = filterValue.trim().toLocaleLowerCase();
+
+	return tags
+		.filter(checkbox => get(checkbox, 'label', '').toLocaleLowerCase().includes(formatFilterValue))
+		.filter(tag => (showAll ? true : tag.checked));
+};
+
+const BadgeTagsForm = ({ tagsValues, id, onChange, onSubmit, value, feature, isLoading, t }) => {
+	const [filter, setFilter] = useState('');
+	const [showAll, setShowAll] = useState(true);
+
+	const badgeTagsFormId = `${id}-tags-form`;
+
+	const tags = useCallback(getTags(tagsValues, value, filter), [tagsValues, value]);
```

Comment:
```suggestion
	const tags = useMemo(() => getTags(tagsValues, value), [tagsValues, value]);
```

There is 2 mistakes here:
* filter is passed but not used in the function, removed it
* useCallback is used to memoize a function. Here you want to memoize a value, not to recompute it if `[tagsValues, value]` don't change. But with useCallback, `getTags()` is always called to get the parameter to pass to useCallback. With useMemo, `getTags()` is only executed when one of the dependencies is changed